frontend/src/pages/Blogs.tsx

```diff
@@ -15,33 +15,68 @@ export interface BlogType {
   claps: [];
   bookmarkId?: string;
 }
-
 const Blogs = () => {
   const { blogs, loading } = useBlogs();
 
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchedData, setSearchedData] = useState([]);
+
+  // Search-Logic
+  useEffect(() => {
+    if (searchQuery) {
+      const debouncedSearch = setTimeout(() => {
+        setSearchedData(
+          blogs.filter((info: any) => {
+            return Object.values(info.title)
+              .join('')
+              .toLowerCase()
+              .includes(searchQuery.toLowerCase());
+          })
+        )
+
+        return () => clearTimeout(debouncedSearch);
+      }, 2000);
+    }
+    else {
+      setSearchedData(blogs);
+    }
+  }, [searchQuery])
+
+
```

Comment:
This needs to be refactored with a custom hook, something like `useSearch` which can return searchQuery, searchedData, setFunc()..